COLA-MVP-version0.3

ZhuMuMu0216 opened the pull request.

ZhuMuMu0216 committed:
version-0.1

ZhuMuMu0216 committed:
version-0.2-push

ZhuMuMu0216 committed:
version-0.3

youlei202 commented on cola/ace_generator.py:
```diff
@@ -0,0 +1,90 @@
+import pandas as pd
+
+class COLA:
+    def __init__(self, ml_model=None, data=None, counterfactual_algorithm=None, policy_name="CF_OTMatch", Avalues_method="max", limited_actions=None):
```
I noticed that you have declared type for other class. I think it is good to do it for all your functions. E.g.

ml_model : Model

youlei202 commented on cola/ace_generator.py:
```diff
@@ -0,0 +1,90 @@
+        # self.ce_algorithm =None
+        # self.results = self.generate_counterfactual()
+
+    def generate_ace_counterfactual(self):
```
generate_action_constrained_counterfactual sounds better.

It is difficult to get the meaning of "ace"

youlei202 commented on cola/ace_generator.py:
```diff
@@ -0,0 +1,90 @@
+        self.ml_model_true = self.ml_model.load_model()
+
+        # get ce_algorithm
+        ce_algorithm = self.counterfactualexplanation_algorithm()
```
counterfactual_explanation_algorithm is better

(I usually do not modify the names that you've chosen, unless it sounds really weird by the standard of python development)

ZhuMuMu0216 replied:
OK, I have already made these changes.

youlei202 commented on cola/ace_generator.py:
```diff
@@ -0,0 +1,90 @@
+        return factual, ce, ace
+
+
+    def counterfactualexplanation_algorithm(self):
```
Later on, you gonna support multiple counterfactual explanations algorithms. I think it is better to explicitly declare as:

def counterfactual_explanation_algorithm(self) -> CounterFactualExplainer:

youlei202 commented on cola/ce_module/base_ce.py:
```diff
@@ -0,0 +1,25 @@
+import pandas as pd
+
+class CounterFactualExplainer(ABC):
+    def __init__(self, ml_model, x_factual: pd.DataFrame):
```
ml_model should also be with a type

ZhuMuMu0216 committed:
comment_revise

ZhuMuMu0216 committed:
version-0.4-push

ZhuMuMu0216 committed:
version-0.5-revise

ZhuMuMu0216 committed:
version-0.6-revise

revise all the small problems( for example, change the input para from explainer into x_factual and x_counterfactual in pshap.py). The next step is to add some more explainer_algorithm and ml_model into the COLA packcage.

ZhuMuMu0216 committed:
version-0.7-alibi_DiCE_AReS_DisCOUNT

Add the alibi, AReS, DiCE, DisCOUNT. Also add 4 new notebooks for testing.
1. DisCount only compatiabl with PYT (need to pick suitable parameters to optimize the best results)
2. AReS algorithm is n to 1(need additional input parameters) - (also have a question about input x_factual, whether can only be integers?)
3. ALIBI is a bit hard to pick the best training parameters.

ZhuMuMu0216 committed:
version-0.8-bigchange

1. delete sample_num
2. change the function generate_counterfactual(), and delete this fucntion in the initial part

ZhuMuMu0216 committed:
version-0.9-revise

1.add some comments
2.complete DICE,DISCOUNT,ALIBI_CFI
3.revise the policy.py

ZhuMuMu0216 committed:
version-0.10-revise

1. make a new wrapper for ml_models: pytorch_model, sklearn_model
2. add two more matcher: "nn", "cem"
3. add some comments for DiCE, DisCount and Alibi-cei

ZhuMuMu0216 committed:
version-0.11-revise

1. merge "dataset" and "rawdata"
2. leave a pre-trained LGBMclassifier_GermanCredit.pkl file

ZhuMuMu0216 committed:
version-0.12-revise

1.重新整理了policy模块
2.整理文件结构

ZhuMuMu0216 committed:
version-0.12-revise

1. almost done (except heatmap.py)
2. ml_model_interface.py (model.predict_proba(X)[:,1] need to be improved)

ZhuMuMu0216 committed:
version-0.13-add_highlight_ace

ZhuMuMu0216 committed:
version-0.14-revise_highlight

1. solve the problem that u can't double run the display() function

ZhuMuMu0216 committed:
version-0.15-heatmap

1.增加heatmap()
2.notebook底部进行了good results筛选

ZhuMuMu0216 committed:
version-0.16-pick_good_results

1.reserve the picked good results for hotel booking dataset
2. notebook contains a lot contents(which will be deleted in next version)

ZhuMuMu0216 committed:
version-0.17-Lite_version

1.删除多余的.ipynb文件
2.只保留最简洁的demo

ZhuMuMu0216 committed:
version-1.0_add_env_readme

1.简易版readme
2.把环境导出了

ZhuMuMu0216 committed:
version-1.0.1-readme_revise

ZhuMuMu0216 committed:
version-1.0.1-readme_CodePart

ZhuMuMu0216 committed:
version-1.0.2-ReadmePart

ZhuMuMu0216 committed:
version-1.0.1-Readme_codepart

ZhuMuMu0216 committed:
version-1.0.2-ReadmeRevise

ZhuMuMu0216 committed:
version-1.0.3-Readme_revise

ZhuMuMu0216 committed:
version-1.0.3-MITlicense

ZhuMuMu0216 committed:
version-1.0.3-arxiv_mitlicense

ZhuMuMu0216 committed:
version-1.0.3-readme

ZhuMuMu0216 committed:
version-1.0.3_README
